src/Internal/SetupEncryptionHandlerV2.cs

```diff
@@ -102,11 +103,13 @@ protected override EncryptionInstructions GenerateInstructions(IExecutionContext
             EncryptionInstructions instructions = null;
             if (NeedToGenerateKMSInstructions(executionContext))
             {
-                instructions = EncryptionUtils.GenerateInstructionsForKMSMaterialsV2(EncryptionClient.KMSClient, EncryptionMaterials);
+                var effectiveEncryptionContext = ValidateAndGetEffectiveEncryptionContext(executionContext, EncryptionMaterials.MaterialsDescription);
```

Comment:
Same question as V1 are you introducing a new exception situation without a user having do something new?